src/commands/exile.py

```diff
@@ -0,0 +1,53 @@
+import discord
+from discord.ext.commands import Bot
+from settings import get_settings
+from services.exile_service import exile_user, unexile_user
+from collections.abc import Coroutine
+
+settings = get_settings()
+
+def create_unexile_command(bot: Bot) -> None:
+    @bot.tree.command()
+    @discord.app_commands.describe(
+        user="User being exiled"
+    )
```

Comment:
i think it should be legal to put this above the function signature. that way we won't have to create a nested function

Reply:
I have the async method nested inside the "create command" method in order to move the creation of commands to their own files, while also being able to pass around the `Bot` instance properly. I think there's a way to simplify this using the `ApplicationCommand` class that discord.py exposes, but I was having some trouble in my first pass of that. I agree that this setup is confusing, but I got the pattern from how [mooncord defines its commands](https://github.com/Shy-16/Mooncord-PitBot/blob/master/application_commands/pit_command.py#L9-L20)